sql/src/main/java/io/crate/expression/scalar/string/TrimFunction.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar.string;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.Chars;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.expression.scalar.TripleScalar;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionInfo;
+import io.crate.types.DataTypes;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.HashSet;
+
+public final class TrimFunction {
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(new TripleScalar<>(functionInfo(), TrimFunction::trim));
+    }
+
+    private static FunctionInfo functionInfo() {
+        return new FunctionInfo(
+            new FunctionIdent("trm", ImmutableList.of(DataTypes.STRING, DataTypes.STRING, DataTypes.STRING)),
+            DataTypes.STRING);
+    }
+
+    @Nullable
+    private static String trim(String mode, String charsToTrim, String value) {
+        if (value == null || charsToTrim == null) {
+            return value;
+        }
+
+        final HashSet<Character> trimmableChars = new HashSet<>(Chars.asList(charsToTrim.toCharArray()));
```

Comment:
@mfussenegger I'm thinking of adding an optimized version for single char strings as I think this use case is heavily used. Do you think it's worth it?

Reply:
I guess it would be worth it. That would mean to initiate a `FunctionCall` with 2 args in the `AstBuilder` so that we can already dispatch to the optimized version instead of having to do a check on `charsToTrim` on runtime for each row, right?

Reply:
I'll be honest - checking this at runtime was what I had in mind :)
But this definitely sounds better 👍

Reply:
Would it also make sense to have an overload with 1 argument to also avoid the `mode` checks?